Src/Support/Google.Apis.Auth.AspNetCore/GoogleCredentialProvider.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Google.Apis.Auth.OAuth2;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Apis.Auth.AspNetCore
+{
+    public class GoogleCredentialProvider : IGoogleCredentialProvider
+    {
+        public GoogleCredentialProvider(IServiceProvider services, string authenticationScheme)
+        {
+            _services = services;
+            _authenticationScheme = authenticationScheme;
+        }
+
+        private readonly IServiceProvider _services;
+        private readonly string _authenticationScheme;
+
+        public async Task<GoogleCredential> GetAsync(CancellationToken cancellationToken)
+        {
+            // TODO: Should failures throw an exception? Or return null (as currently coded)?
+            var httpContext = _services.GetRequiredService<IHttpContextAccessor>().HttpContext;
+            var auth = await httpContext.AuthenticateAsync(_authenticationScheme);
+            if (!auth.Succeeded || auth.None)
+            {
+                // Auth failed, cannot provide a credential.
+                // TODO: Logging
+                return null;
+            }
+            var optionsMonitor = _services.GetRequiredService<IOptionsMonitor<OpenIdConnectOptions>>();
```

Comment:
I've tried getting this to work (in a slightly different setting) but for me this only works if you specifically request Google options, so:

`var optionsMonitor = _services.GetRequiredService<IOptionsMonitor<GoogleOptions>>();`

Are you sure this works here?

Reply:
By the way, @chrisdunelm I'd love to see your working example of this. I could use it right away in this project that I'm working on. I think your approach is how this should be done, but I wasn't able to get it to work.

Reply:
Somewhat later (more hours than I'd like to admit) I got this to work flawlessly. Thanks @chrisdunelm